src/Symfony/Component/Routing/RouteCollection.php

```diff
@@ -54,7 +54,7 @@ public function __clone()
      *
      * @see all()
      *
-     * @return \ArrayIterator|Route[] An \ArrayIterator object for iterating over routes
+     * @return \ArrayIterator<Route>
```

Comment:
> ArrayIterator has missing template params, expecting 2
Psalm requires `ArrayIterator` to define `<key type, value type>`

> Extended template param TKey of ArrayIterator<Symfony\Component\Routing\Route> expects type array-key, type Symfony\Component\Routing\Route given

It requires the `key type` to have `array-key` as upper type (i.e. `string` or `int`), so it errors because you've now set it to a `Route`object.

> The declared return type 'ArrayIterator<Symfony\Component\Routing\Route>' for Symfony\Component\Routing\RouteCollection::getIterator is incorrect, got 'ArrayIterator<array-key, Symfony\Component\Routing\Route>'

And finally it's saying that `ArrayIterator<array-key, Route>` is actually returned, which is probably also what you meant by this PHPdoc.
```suggestion
     * @return \ArrayIterator<array-key,Route>
```

Reply:
thanks! updated